test/unit/test-amp-story-player.js

```diff
@@ -691,58 +689,135 @@ describes.realWin('AmpStoryPlayer', {amp: false}, (env) => {
+      player.go(4);
+      swipeLeft();
+      expect(navigationSpy).to.have.been.calledWithMatch({
+        type: 'navigation',
+        detail: {
+          index: 4,
+          remaining: 0,
+        },
       });
     });
 
-    it('signals when player changed story using previous method', async () => {
+    it('first story call to previous_() is the last story', async () => {
       const playerEl = win.document.createElement('amp-story-player');
       appendStoriesToPlayer(playerEl, 5);
 
       const player = new AmpStoryPlayer(win, playerEl);
 
       await player.load();
+      await nextTick();
 
       const navigationSpy = env.sandbox.spy();
       playerEl.addEventListener('navigation', navigationSpy);
-      player.next_();
-      player.previous_();
-      expect(navigationSpy.secondCall.args[0].type).to.eql('navigation');
-      expect(navigationSpy.secondCall.args[0].detail).to.eql({
-        index: 0,
-        remaining: 4,
+      playerEl.setAttribute('enable-circular-wrapping', '');
+      player.go(5);
+      swipeRight();
+      expect(navigationSpy).to.have.been.calledWithMatch({
+        type: 'navigation',
+        detail: {
+          index: 0,
+          remaining: 4,
+        },
       });
     });
 
-    it('signals when player changed story using go method', async () => {
+    it('navigate to first story when last story is finished', async () => {
       const playerEl = win.document.createElement('amp-story-player');
       appendStoriesToPlayer(playerEl, 5);
 
       const player = new AmpStoryPlayer(win, playerEl);
 
       await player.load();
+      await nextTick();
 
       const navigationSpy = env.sandbox.spy();
       playerEl.addEventListener('navigation', navigationSpy);
+      playerEl.setAttribute('enable-circular-wrapping', '');
+      player.go(4);
       player.go(1);
-      expect(navigationSpy.firstCall.args[0].type).to.eql('navigation');
-      expect(navigationSpy.firstCall.args[0].detail).to.eql({
-        index: 1,
-        remaining: 3,
+      expect(navigationSpy).to.have.been.calledWithMatch({
+        type: 'navigation',
+        detail: {
+          index: 0,
+          remaining: 4,
+        },
+      });
+    });
+
+    it('navigate to last story when first story is requested to go back', async () => {
+      const playerEl = win.document.createElement('amp-story-player');
+      appendStoriesToPlayer(playerEl, 5);
+
+      const player = new AmpStoryPlayer(win, playerEl);
+
+      await player.load();
+      await nextTick();
+
+      const navigationSpy = env.sandbox.spy();
+      playerEl.addEventListener('navigation', navigationSpy);
+      playerEl.setAttribute('enable-circular-wrapping', '');
+      player.go(-1);
+      expect(navigationSpy).to.have.been.calledWithMatch({
+        type: 'navigation',
+        detail: {
+          index: 4,
+          remaining: 0,
+        },
+      });
+    });
+
+    it('signals when player changed story using next method', async () => {
```

Comment:
Please delete this and the next test, since we are already testing all the navigation options that use `next_()` and `previous_()`, and we don't want to use private methods in unit tests.